### PR TITLE
feat: enable telemetry events

### PR DESCRIPTION
**Pull request by alex-gilin**

### CODE REVIEW

File: packages/tasks_panel/src/commands/edit-task.ts [hunk 1]
```diff
@@ -23,12 +23,6 @@ export async function editTreeItemTask(
 }
 
 async function editTask(task: ConfiguredTask, readResource: (file: string) => Promise<string>): Promise<void> {
```
Comment (veredcon) [inline packages/tasks_panel/src/commands/edit-task.ts:25]: edit task report is not interesting to report?

File: packages/tasks_panel/src/usage-report/usage-analytics-wrapper.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    AnalyticsWrapper.report(AnalyticsWrapper.EVENT_TYPES.TASK_CREATE_FINISHED, {
+      intent: properties.taskType ?? "",
+      type: properties.type,
+    });
+  }
+
+  public static reportTaskSave(properties: Properties): void {
+    AnalyticsWrapper.report(AnalyticsWrapper.EVENT_TYPES.TASK_SAVE, {
+      intent: properties.__intent,
+      extensionName: properties.__extensionName,
```
Comment (veredcon) [inline packages/tasks_panel/src/usage-report/usage-analytics-wrapper.ts:79]: `extensionName` is already being mentioned as part of the event name so maybe no need to add it as property also

File: packages/tasks_panel/src/usage-report/usage-analytics-wrapper.ts [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+  public static reportTaskCreateFinished(properties: Properties): void {
+    AnalyticsWrapper.report(AnalyticsWrapper.EVENT_TYPES.TASK_CREATE_FINISHED, {
+      intent: properties.taskType ?? "",
+      type: properties.type,
+    });
+  }
+
+  public static reportTaskSave(properties: Properties): void {
+    AnalyticsWrapper.report(AnalyticsWrapper.EVENT_TYPES.TASK_SAVE, {
+      intent: properties.__intent,
```
Comment (veredcon) [inline packages/tasks_panel/src/usage-report/usage-analytics-wrapper.ts:78]: is it clear what `intent` is?

**Review by veredcon — APPROVED** [review]

File: packages/tasks_panel/src/commands/edit-task.ts (hunk 1 above)
Comment (alex-gilin) [reply packages/tasks_panel/src/commands/edit-task.ts:25]: there is a "task save" event that covers this

File: packages/tasks_panel/src/usage-report/usage-analytics-wrapper.ts (hunk 2 above)
Comment (alex-gilin) [reply packages/tasks_panel/src/usage-report/usage-analytics-wrapper.ts:79]: in that case "extensionName" is s name of task provider extension but not a task-explorer itself.

File: packages/tasks_panel/src/usage-report/usage-analytics-wrapper.ts (hunk 3 above)
Comment (alex-gilin) [reply packages/tasks_panel/src/usage-report/usage-analytics-wrapper.ts:78]: this is an internal term, but in the report it means "task group" (displayed as "build", "deploy", etc.)

**Pull request merged**